src/GovUk.Education.ExploreEducationStatistics.Data.Processor/ImportRecoveryHandler.cs

```diff
@@ -66,19 +66,24 @@ public static void CheckIncompleteImports(string connectionString)
                     // Else create a message for all remaining batches
                     else
                     {
+                        Console.WriteLine($"Creating messages for remaining batches for {entity.PartitionKey} : {entity.RowKey}");
                         var m = JsonConvert.DeserializeObject<ImportMessage>(entity.Message);
                         var batches = GetBatchesRemaining(blobContainer, entity.PartitionKey, m.OrigDataFileName)
                             .Result;
 
                         // If no batches then assume it didn't get passed initial validation stage
                         if (!batches.Any())
                         {
+                            Console.WriteLine($"No batches created for {entity.PartitionKey} : {entity.RowKey}`` although batches are necessary to create.  Adding message to the queue to try re-import.");
                             pendingQueue.AddMessage(new CloudQueueMessage(entity.Message));
-                            return;
+                            continue;
```

Comment:
Ah the empty queues make a lot of sense now!

Reply:
:)